cli/command/volume/create.go

```diff
@@ -48,16 +67,130 @@ func newCreateCommand(dockerCli command.Cli) *cobra.Command {
 	flags.VarP(&options.driverOpts, "opt", "o", "Set driver specific options")
 	flags.Var(&options.labels, "label", "Set metadata for a volume")
 
+	// flags for cluster volumes only
+	flags.StringVar(&options.group, "group", "", "Cluster Volume group (cluster volumes)")
+	flags.SetAnnotation("group", "version", []string{"1.42"})
+	flags.SetAnnotation("group", "swarm", []string{"manager"})
+	flags.StringVar(&options.scope, "scope", "single", `Cluster Volume access scope ("single"|"multi")`)
+	flags.SetAnnotation("scope", "version", []string{"1.42"})
+	flags.SetAnnotation("scope", "swarm", []string{"manager"})
+	flags.StringVar(&options.sharing, "sharing", "none", `Cluster Volume access sharing ("none"|"readonly"|"onewriter"|"all")`)
+	flags.SetAnnotation("sharing", "version", []string{"1.42"})
+	flags.SetAnnotation("sharing", "swarm", []string{"manager"})
+	flags.StringVar(&options.availability, "availability", "active", `Cluster Volume availability ("active"|"pause"|"drain")`)
+	flags.SetAnnotation("availability", "version", []string{"1.42"})
+	flags.SetAnnotation("availability", "swarm", []string{"manager"})
+	flags.StringVar(&options.accessType, "type", "block", `Cluster Volume access type ("mount"|"block")`)
+	flags.SetAnnotation("type", "version", []string{"1.42"})
+	flags.SetAnnotation("type", "swarm", []string{"manager"})
+	flags.Var(&options.secrets, "secret", "Cluster Volume secrets")
+	flags.SetAnnotation("secret", "version", []string{"1.42"})
+	flags.SetAnnotation("secret", "swarm", []string{"manager"})
+	flags.Var(&options.limitBytes, "limit-bytes", "Minimum size of the Cluster Volume in bytes")
+	flags.SetAnnotation("limit-bytes", "version", []string{"1.42"})
+	flags.SetAnnotation("limit-bytes", "swarm", []string{"manager"})
+	flags.Var(&options.requiredBytes, "required-bytes", "Maximum size of the Cluster Volume in bytes")
+	flags.SetAnnotation("required-bytes", "version", []string{"1.42"})
+	flags.SetAnnotation("required-bytes", "swarm", []string{"manager"})
+	flags.Var(&options.requisiteTopology, "topology-required", "A topology that the Cluster Volume must be accessible from")
+	flags.SetAnnotation("topology-required", "version", []string{"1.42"})
+	flags.SetAnnotation("topology-required", "swarm", []string{"manager"})
+	flags.Var(&options.preferredTopology, "topology-preferred", "A topology that the Cluster Volume would be preferred in")
+	flags.SetAnnotation("topology-preferred", "version", []string{"1.42"})
+	flags.SetAnnotation("topology-preferred", "swarm", []string{"manager"})
+
 	return cmd
 }
 
+// hasClusterVolumeOptionSet returns true if any of the cluster-specific
+// options are set.
+func hasClusterVolumeOptionSet(flags *pflag.FlagSet) bool {
+	return flags.Changed("group") || flags.Changed("scope") ||
+		flags.Changed("sharing") || flags.Changed("availability") ||
+		flags.Changed("type") || flags.Changed("secrets") ||
+		flags.Changed("limit-bytes") || flags.Changed("required-bytes")
```

Comment:
Yes, there's some things to clean up in this area. I had a variant where I defined a slice of flag-names that are related, and used the slice for both setting the annotations and for this code, but in the end decided to keep my changes on top of @dperny's PR minimal

My goal with this rebase was to have  the PR in a "mergeable" state, so that I could do a test-build/test-release of 22.06 with at least _something_ that exposes the new feature. So I only fixed the type renames from upstream, and added the second commit to have the correct annotations on the flags.